### PR TITLE
Keep go-containerregistry up to date

### PR DESCRIPTION
**Pull request by lcarva**
Ref: EC-778

### CODE REVIEW

**Review by simonbaird — APPROVED** [review]
I didn't look too closely, but should be fine to merge early and often.

**Comment by lcarva** [comment]
That didn't quite work. But at least now I can try things out [here](https://github.com/enterprise-contract/ec-cli/compare/fix-app-id?expand=1).

Current failure: https://github.com/enterprise-contract/ec-cli/actions/runs/11002493835/job/30549573454

I thought that was due to a bad app ID, but AFAICT, I am now providing the right one. I'll keep digging.

**Comment by lcarva** [comment]
It works!

https://github.com/enterprise-contract/ec-cli/pull/2007

https://github.com/enterprise-contract/ec-cli/actions/runs/11002747801/job/30550367225

But! It didn't trigger any of the checks... It's probably missing some permissions somewhere.

File: .github/workflows/update-go-containerregistry.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - name: update
+        run: |
+          go mod edit -replace github.com/google/go-containerregistry=github.com/enterprise-contract/go-containerregistry@main
+          go mod tidy
+        env:
+          # Hack to ensure this repo is always updated, see:
+          #   https://github.com/golang/go/issues/45413
+          GOPRIVATE: github.com/enterprise-contract/go-containerregistry
+
+      - uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 # v1.11.0
```
Comment (zregvart) [inline .github/workflows/update-go-containerregistry.yaml:54]: TIL

**Comment by lcarva** [comment]
> But! It didn't trigger any of the checks... It's probably missing some permissions somewhere.

Yeah because I ran the workflow from a branch (fix-app-id) so the PR was created targeting that branch. Since our CI mostly targets the `main` branch, there was nothing to run.

**Pull request merged**